cpp/libs/gst_nmos_sender_plugin/gst_nmos_sender_plugin.cpp

```diff
@@ -0,0 +1,442 @@
+        g_object_set(G_OBJECT(self->udpsink), "host", self->config.network.destination_address, NULL);
+        break;
+
+    case PropertyId::DestinationPort:
+        self->config.network.destination_port = atoi(g_value_dup_string(value));
+        g_object_set(G_OBJECT(self->udpsink), "port", self->config.network.destination_port, NULL);
+        break;
+
+    default: G_OBJECT_WARN_INVALID_PROPERTY_ID(object, property_id, pspec); break;
+    }
+}
+
+static void gst_nmossender_get_property(GObject* object, guint property_id, GValue* value, GParamSpec* pspec)
+{
+    GstNmossender* self = GST_NMOSSENDER(object);
+
+    switch(static_cast<PropertyId>(property_id))
+    {
+    case PropertyId::NodeId: g_value_set_string(value, self->config.node.id); break;
+    case PropertyId::NodeConfigFileLocation: g_value_set_string(value, self->config.node.configuration_location); break;
+    case PropertyId::DeviceId: g_value_set_string(value, self->config.device.id); break;
+    case PropertyId::DeviceLabel: g_value_set_string(value, self->config.device.label); break;
+    case PropertyId::DeviceDescription: g_value_set_string(value, self->config.device.description); break;
+    case PropertyId::SenderId: g_value_set_string(value, self->config.sender_id); break;
+    case PropertyId::SenderLabel: g_value_set_string(value, self->config.sender_label); break;
+    case PropertyId::SenderDescription: g_value_set_string(value, self->config.sender_description); break;
+    case PropertyId::SourceAddress: g_value_set_string(value, self->config.network.source_address); break;
+    case PropertyId::InterfaceName: g_value_set_string(value, self->config.network.interface_name); break;
+    case PropertyId::DestinationAddress: g_value_set_string(value, self->config.network.destination_address); break;
+    case PropertyId::DestinationPort: g_value_set_int(value, self->config.network.destination_port); break;
+
+    default: G_OBJECT_WARN_INVALID_PROPERTY_ID(object, property_id, pspec); break;
+    }
+}
+
+/* Event handler for the sink pad */
+static gboolean gst_nmossender_sink_event(GstPad* pad, GstObject* parent, GstEvent* event)
+{
+    GstNmossender* self = GST_NMOSSENDER(parent);
+
+    switch(GST_EVENT_TYPE(event))
+    {
+    case GST_EVENT_CAPS: {
+        GstCaps* caps = NULL;
+        gst_event_parse_caps(event, &caps);
+
+        if(caps)
+        {
+            gchar* caps_str = gst_caps_to_string(caps);
+            g_print("\n\nReceived CAPS from upstream: %s\n", caps_str);
+
+            const GstStructure* structure = gst_caps_get_structure(caps, 0);
+            const gchar* media_type       = gst_structure_get_name(structure);
+
+            if(g_strcmp0(media_type, "audio/x-raw") == 0)
+            {
+                // DEBUG
+                g_print("Audio caps detected\n");
+                GST_INFO_OBJECT(self, "Audio caps detected");
+                //
+                gst_element_link(self->queue, self->audio_payloader);
```

Comment:
The `gst_element_link` returns false if it can't link the elements. So we must check the return of the function. 
There are other GStreamer functions that we must check for the return, for example:
- `gst_element_get_static_pad` returns NULL if it is unable to get the pad
- `gst_element_add_pad` can return false if a pad with the same name already exists

So please check GStreamer functions returns and that we safely exit in case of an error